Ar_MethodHw.js

```diff
@@ -0,0 +1,185 @@
+// // 1.
+//
+Array.prototype.myForEach = function(callback) {
+  for (let i = 0; i < this.length; i++) {
+    callback(this[i]);
+  }
+};
+
+//
+// //[1, 2, 3].myForEach((el) => {
+//   //console.log(el)
+// //})
+// // 2.
+
+// })
+// SOLUTION USING A LOOP INSTEAD
+// Array.prototype.myMap = function(callback) {
+//  let result =[];
+//  for(let i = 0; i < this.length; i++) {
+//    result.push(callback(this[i]))
+//
+//  }
+//  return result
+//  console.log(result)
+// };
+//
+// Array.prototype.myMap = function(anyT){
+//    let result = [];
+//    this.myForEach(element => {
+//
+//      result.push(anyT(element));
+//    })
+//
+//    return result;
+// }
+//
+//
+// let double = [1, 2, 3].myMap((el) => {
+//  return el * 2;
+// })
+// console.log(double);
+// 3.
+// Array.prototype.myFilter = function(callback){
+//     let result = [];
+//     this.myForEach(element => {
+//
+//       result.push(callback(element));
```

Comment:
Remember, we don't actually want to use the callback to change the element. We want to use it to determine *whether or not* it should be pushed into result.